srcCpp03/PerftestTransport.cxx

```diff
@@ -332,42 +415,160 @@ void configureShmemTransport(
+        ss.str("");
+        ss.clear();
+        ss << receiveBufferSize;
+        qos_properties["dds.transport.shmem.builtin.receive_buffer_size"] = ss.str();
+        transport.loggingString +=
+                ("\tSHMEM receive_buffer_size: "
+                + ss.str()
+                + "\n");
+    }
+}
+
+/*
+ * Gets the MessageSizeMax given the name (String) of a transport from a
+ * DDS_DomainParticipantQos object. If the value is not present, returns
+ * DEFAULT_MESSAGE_SIZE_MAX.
+ */
+long getTransportMessageSizeMax(
+        std::string targetTransportName,
+        PerftestTransport &transport,
+        std::map<std::string, std::string> &qos_properties)
+{
+    std::string propertyName =
+            transport.transportConfigMap[targetTransportName].prefixString
+            + ".parent.message_size_max";
+
+    if (qos_properties.find(propertyName) != qos_properties.end()) {
+        //printf("Value for %s is %s\n", propertyName.c_str(), qos_properties[propertyName].c_str());
+        return atoi(qos_properties[propertyName].c_str());
+    } else {
+        //printf("Value for %s not found, returning default\n", propertyName.c_str());
+        return DEFAULT_MESSAGE_SIZE_MAX;
+    }
+}
+
+/*
+ * Configures the minimumMessageSizeMax value in the PerftestTransport object with
+ * the minimum value for all the enabled transports in the XML configuration.
+ */
+void getTransportMinimumMessageSizeMax(
+        PerftestTransport &transport,
+        dds::domain::qos::DomainParticipantQos &qos,
+        std::map<std::string, std::string> &qos_properties)
+{
+    using namespace rti::core::policy;
+
+    long qosConfigurationMessageSizeMax = MESSAGE_SIZE_MAX_NOT_SET;
+    long transportMessageSizeMax = MESSAGE_SIZE_MAX_NOT_SET;
+    TransportBuiltinMask mask = qos.policy<TransportBuiltin>().mask();
+
+    if ((mask & TransportBuiltinMask::shmem()) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax("SHMEM", transport, qos_properties);
+
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if ((mask & TransportBuiltinMask::udpv4()) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax(
+                "UDPv4",
+                transport,
+                qos_properties);
+
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if ((mask & TransportBuiltinMask::udpv6()) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax(
+                "UDPv6",
+                transport,
+                qos_properties);
+
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if (transport.transportConfig.kind == TRANSPORT_TCPv4
```

Comment:
These three (TCP,TLS and DTL) should be if-else statements.

Reply:
True. Fixed.

Reply:
Same for srcCpp